akka-docs/src/main/paradox/scala/general/actor-systems.md

```diff
@@ -107,14 +107,153 @@ which is a single point of contention if over-used.
+unless you @ref:[set up a separate dispatcher for the actor](../dispatchers.md).
+
+If all of the available threads are blocked, then all the actors on the same dispatchers will starve for threads and
+they cannot process any more incoming message.
+
+@@@ note
+
+Blocking APIs should also be avoided if possible. Try to find or build Reactive APIs,
+such that blocking is minimised, or moved over to dedicated dispatchers.
+
+Often when integrating with existing libraries or systems it is not possible to
+avoid blocking APIs. The following solution explains how to handle blocking
+operations properly.
+
+Note that the same hints apply to managing blocking operations anywhere in Akka,
+including in Actors etc.
+
+@@@
+
+### Problem example: blocking the default dispatcher
+
+Let's set up an application with the above `BlockingFutureActor` and the following `PrintActor`. 
+
+@@snip [BlockingDispatcherSample.scala](../../../../test/scala/docs/actor/BlockingDispatcherSample.scala) { #print-actor }
+
+@@snip [BlockingDispatcherSample.scala](../../../../test/scala/docs/actor/BlockingDispatcherSample.scala) { #blocking-main }
+
+Here the app is sending 100 messages to `BlockingFutureActor` and `PrintActor` and large numbers
+of akka.actor.default-dispatcher threads are handling requests. When you run run the above code,
+you will likely to see the entire application gets stuck somewhere like this:
+
+```
+>　PrintActor: 44
+>　PrintActor: 45
+```
+
+(i.e.) (Ignoring `println`'s latency) `PrintActor` is non-blocking but it cannot process all the 100 messages asap.
+
+In the thread state diagrams below the colours have the following meaning:
+
+ * Turquoise - Sleeping state
+ * Orange - Waiting state
+ * Green - Runnable state
+
+The thread information was recorded using the YourKit profiler, however any good JVM profiler 
+has this feature (including the free and bundled with the Oracle JDK VisualVM, as well as Oracle Flight Recorder). 
+
+The orange portion of the thread shows that it is idle. Idle threads are fine -
+they're ready to accept new work. However, large amounts of Turquoise (sleeping) threads are very bad!
+
+![dispatcher-behaviour-on-good-code.png](../../images/dispatcher-behaviour-on-bad-code.png)
+
+The app is exposed to the load of `i: Int` messages which will block these threads. 
+For example "`default-akka.default-dispatcher2,3,4`"
+are going into the blocking state, after having been idle. It can be observed
+that the number of new threads increases, "`default-akka.actor.default-dispatcher 18,19,20,...`" 
+however they go to sleep state immediately, thus wasting the
+resources.
+
+The number of such new threads depends on the default dispatcher configuration,
+but it will likely not exceed 50. Since many requests are being processed, the entire
+thread pool is starved. The blocking operations dominate threads in the dispatcher 
+such that it has no thread available to handle other requests.
```

Comment:
Remove this paragraph, the above rewrite includes that info, and also let's avoid calling out magic numbers since that could change depending on config